src/store/reducers/storage/utils.ts

```diff
@@ -243,10 +245,22 @@ export const calculateMaximumSlotsPerDisk = (
     );
 };
 
+export const calculateMaximumDisksPerNode = (
+    nodes: TNodeInfo[] | undefined,
+    providedMaximumDisksPerNode?: string,
+) => {
+    if (providedMaximumDisksPerNode) {
+        return providedMaximumDisksPerNode;
+    }
+
+    return String(Math.max(1, ...(nodes || []).map((node) => node.PDisks?.length || 0)));
```

Comment:
it's quite hard to read. Could you please refactor?

Reply:
mission accomplished

mind that its a fallback function for corner case of a corner case